### ISSUE TITLE
Implement pipe-composition

### ISSUE THREAD

**Issue by ELLIOTTCABLE** [ELLIOTTCABLE/excmd.js#5]
This needs discussion, though I haven't thought about it in detail yet. Obviously the existing hack using `composite` can be done away with entirely. Less-obviously, I *think* we aren't going to see any of the problems that Vim's excmds have with the pipe — i.e. needing to conditionally support pipe-as-composition with *some* commands, but not others?

Thoughts and feedback welcome. Probably going to throw together naive, `command | another_command` support pretty soon; and can develop edge-cases from there.

This is one place where good examples of real-world uses of `composite` would be *great*.

**Comment by ELLIOTTCABLE** [comment]
Oh crap, search terms. This may require more thought …

**Comment by glacambre** [comment]
Yeah, search terms are going to be annoying. There's also `js if (something || other)` and `exclaim cat file | grep pattern` that will probably make things harder.

Here are the composite commands I can find in my .tridactylrc:
```
composite js Array.from(document.querySelectorAll("li.list-group-item > span:nth-child(1) > a:nth-child(2)")).map(e => e.href) | jsb -p JS_ARG.forEach(url => tri.excmds.tabopen(url))
composite url2args | fillcmdline open
alias gitclone jsb -p tri.native.run("git clone '" + JS_ARG + "' /home/me/prog/" + JS_ARG.split("/").slice(-1))
composite js window.location.href | gitclone
```

**Comment by ELLIOTTCABLE** [comment]
Awesome! I was planning to come ask y'all for a corpus of example commands 🤣 

For bind, I can't believe I haven't written it down anywhere, but I guess I was quietly assuming we'd support parenthetical disambiguation of expressions of some form! For example:

    bind key (command1 | command2)
    // vs.
    bind key command1 | command2

**Comment by ELLIOTTCABLE** [comment]
Oh, and I explicitly, strongly, very very very strongly do not want to try to support complicated usages of `js` or `exclaim` as simple commands, without quoting, by the way. That `Array.from` composite is a monster! See #7 — but I'm sure you can see all the problems off the top of your head. Something like `js window.location.href | gitclone` looks fine, though?

I'm sure we can add some interim heuristics on the `exclaim` and `js` commands, early on, providing user-friendly, excellent feedback on that breaking change — but we definately shouldn't support it moving forward.
